### PR TITLE
[test] Add %target-run-simple-swiftgyb substitution

### PR DESCRIPTION
**Pull request by rintaro**
#### What's in this pull request?

Add `%target-run-simple-swiftgyb` substitution which will be expended to:

```
rm -rf %t && \
mkdir -p %t && \
%gyb %s -o %t/main.swift && \
%line-directive %t/main.swift -- %target-build-swift %t/main.swift -o %t/a.out -module-name main && \
%line-directive %t/main.swift -- %target-run %t/a.out
```

As ground works:
- GYB: Added `__file__` binding to templates.
- Adjusted `validation-test/stdlib/{Collection,Slice}`
  - Modified `Input/Template.swift.gyb` `utils/split_file.py` compatible
  - Since result scripts are not gyb file, use `%target-run-simple-swift` substitution
  - Renamed `validation-test/stdlib/Slice/*.swift.gyb` to `validation-test/stdlib/Slice/*.swift`

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

### CODE REVIEW

File: test/lit.cfg [hunk 1]
```diff
@@ -641,6 +641,11 @@ if run_vendor == 'apple':
                 "%s %%s" % (target_run_base))
             config.target_run_stdlib_swift = (
                 "%s -Xfrontend -disable-access-control %%s" % (target_run_base))
+            config.target_run_simple_swiftgyb = (
+                'rm -rf %%t && mkdir -p %%t && '
+                '%%gyb %%s -o %%t/main.swift && '
+                '%%line-directive %%t/main.swift -- %s %%t/main.swift'
```
Comment (gribozavr) [inline test/lit.cfg]: Is the build step somehow missing here?

File: test/lit.cfg [hunk 2]
```diff
@@ -721,6 +726,11 @@ elif run_os == 'linux-gnu' or run_os == 'linux-gnueabihf' or run_os == 'freebsd'
             '%s %%s' % (target_run_base))
         config.target_run_stdlib_swift = (
             '%s -Xfrontend -disable-access-control %%s' % (target_run_base))
+        config.target_run_simple_swiftgyb = (
+            'rm -rf %%t && mkdir -p %%t && '
+            '%%gyb %%s -o %%t/main.swift && '
+            '%%line-directive %%t/main.swift -- %s %%t/main.swift'
+            % (target_run_base))
```
Comment (gribozavr) [inline test/lit.cfg]: And here.

File: validation-test/stdlib/Collection/Inputs/Template.swift.gyb [hunk 3]
```diff
@@ -54,19 +54,15 @@ def test_methods(test):
 }%
 
 % for test in all_tests():
-// Start of file: ${test.name}
+// BEGIN ${test.name}
```
Comment (gribozavr) [inline validation-test/stdlib/Collection/Inputs/Template.swift.gyb]: Oh, this file was compatible with `utils/split-generated-tests`.  Unfortunate duplication with `utils/split_file.py`, I agree, but `utils/split-generated-tests` was filtering out `###sourceLocation` markers.

**Comment by gribozavr** [comment]
@swift-ci Please test

File: test/lit.cfg (hunk 1 above)
Comment (rintaro) [inline test/lit.cfg]: This is for `interpret` mode: I think `swift` command would build and run the script.

**Comment by gribozavr** [comment]
@swift-ci Please smoke test

**Comment by gribozavr** [comment]
@rintaro This is a **massive** simplification, thank you!

File: validation-test/stdlib/Collection/Inputs/Template.swift.gyb (hunk 3 above)
Comment (rintaro) [inline validation-test/stdlib/Collection/Inputs/Template.swift.gyb]: Oh, I didn't aware `utils/split-generated-tests`.
`gyb` supports line-directive less output: `--line-directive=""`.

File: test/lit.cfg (hunk 1 above)
Comment (gribozavr) [inline test/lit.cfg]: I see -- you're absolutely right.

File: validation-test/stdlib/Collection/Inputs/Template.swift.gyb (hunk 3 above)
Comment (gribozavr) [inline validation-test/stdlib/Collection/Inputs/Template.swift.gyb]: Good point!  Feel free to remove `utils/split-generated-tests` then or merge it into `utils/split_file.py`.

File: validation-test/stdlib/Slice/MutableRangeReplaceableRandomAccessSlice_Of_MinimalMutableRangeReplaceableRandomAccessCollection_WithPrefixAndSuffix.swift [hunk 4]
```diff
@@ -116,3 +112,4 @@ SliceTests.addMutableRandomAccessCollectionTests(
 )
 
 runAllTests()
+
```
Comment (rintaro) [inline validation-test/stdlib/Slice/MutableRangeReplaceableRandomAccessSlice_Of_MinimalMutableRangeReplaceableRandomAccessCollection_WithPrefixAndSuffix.swift]: This last new line is undesirable change. will fix.

**Comment by gribozavr** [comment]
@swift-ci Please test

**Comment by gribozavr** [comment]
@swift-ci Please smoke test

**Comment by gribozavr** [comment]
@rintaro I'm sorry -- swiftpm build seems to be broken, which blocks running Swift tests.  I'm afraid of merging a large change during this time, I'd prefer to wait until the master branch is more healthy.

**Comment by rintaro** [comment]
Confirmed recent change doesn't need this substitution.
To make sure it will cleanly merge, I rebased on the master.

@swift-ci Please test

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Pull request merged**